projects/common-platform-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/service/PersonService.kt

```diff
@@ -75,8 +81,35 @@ class PersonService(
 
             val savedEquality = equalityRepository.save(equality)
 
-            val savedAddress =
-                defendant.personDefendant.personDetails.address.takeIf { it.containsInformation() }?.let {
+            val addressInfo = defendant.personDefendant.personDetails.address
+            val osPlacesResponse = addressInfo?.takeIf { it.containsInformation() && !it.postcode.isNullOrBlank() }
+                ?.let { findAddressByFreeText(it) }
+
+            val deliveryPointAddress = osPlacesResponse?.results?.firstOrNull()?.dpa
+            val isValidMatch = deliveryPointAddress?.match?.let { it >= 0.6 } ?: false
```

Comment:
Could we use the `minmatch` parameter on the API for this, instead of pulling back everything and filtering on our side?